dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -521,32 +547,37 @@ private boolean createWorkflowsAndVersionsFromDockstoreYml(List<? extends Workfl
                         Workflow workflow = createOrGetWorkflow(workflowType, repository, user, workflowName, wf, gitHubSourceCodeRepo);
                         WorkflowVersion version = addDockstoreYmlVersionToWorkflow(repository, gitReference, dockstoreYml, gitHubSourceCodeRepo, workflow, defaultVersion, yamlAuthors);
 
-                        addValidationsToMessage(workflow, version, messageWriter);
-                        publishWorkflowAndLog(workflow, publish, user, repository, gitReference);
+                        LambdaEvent lambdaEvent = createBasicEvent(repository, gitReference, username, LambdaEvent.LambdaEventType.PUSH, true, lambdaEventGroupId, computeWorkflowName(wf));
```

Comment:
> This method can ultimately be called from io.dockstore.webservice.resources.WorkflowResource#handleGitHubInstallation, in which case the lambda event type is an `INSTALL`, not a `PUSH`.
> 
> This problem existed prior to your PR.

This might be ok, there'll be an INSTALL event created elsewhere, and then some PUSHes here, corresponding to the entries from the `.dockstore.yml` on each of the discovered branches (maybe some PUBLISH events, too!)

Reply:
> This might be ok, there'll be an INSTALL event created elsewhere, and then some PUSHes here, corresponding to the entries from the `.dockstore.yml` on each of the discovered branches (maybe some PUBLISH events, too!)

I don't feel super-strongly about it, and most of the time it may not matter, but it still seems like it would be confusing to have an event that says there was a push when there was no push.

Reply:
> I don't feel super-strongly about it, and most of the time it may not matter, but it still seems like it would be confusing to have an event that says there was a push when there was no push.

One thing we might consider doing eventually is to adjust our terminology and possibly our event scheme.  After this PR, a PUSH event won't correspond to a full push anymore, but rather the creation/update of a single entry.  Might be as easy as relabelling the PUSH event as UPDATEs in the UI event log (or maybe more accurately, UPSERTs, although not sure most people know what that means), or something like that.

Reply:
(This would also dovetail if we someday allow a `.dockstore.yml` repo to be manually refreshed, to support the firewalled on-prem scheme, for example.  There's no "push" there, just some upserts in response to the refresh...)

Reply:
> maybe some PUBLISH events, too!

I lean towards keeping it as is because they'll all have the same delivery ID. I think in some scenarios it would be confusing if they were all INSTALLs.

Example: The install event releases workflowA and also publishes it. The user views the GitHub app logs and sees two lambda events for workflowA and they both say INSTALL instead of one PUSH and one PUBLISH

Reply:
OK fair point about the Publish event, which Steve had made earlier as well, but I'd glossed over.